tests/conftest.py

```diff
@@ -55,22 +55,21 @@ def pytest_pyfunc_call(pyfuncitem):
     if not is_gui_test(pyfuncitem):
         return
 
-    exc_info = None
+    fut = futures.Future()
 
     def testfunc():
+        nonlocal fut
         try:
-            # Call actual test function
-            return pyfuncitem.obj()
-        except Exception:
-            nonlocal exc_info
-            exc_info = sys.exc_info()
+            result = pyfuncitem.obj()
+            fut.set_result(result)
+        except Exception as e:
+            fut.set_exception(e)
 
-    with futures.ThreadPoolExecutor(max_workers=1) as executor:
-        fut = executor.submit(testfunc)
-        fut.result(timeout=GUI_TEST_TIMEOUT)
+    runner = threading.Thread(target=testfunc)
+    runner.daemon = True
+    runner.start()
```

Comment:
For me it's not obvious when will this thread be killed...

Reply:
This tread will be killed either when the test function completes or if it hangs it will be killed on process exit (this is why it's a daemon thread).